Bump up AWS & Terraform Requirements

baalajimaestro opened the pull request.

baalajimaestro committed:
Migrate map() to {}

map() is deprecated since terraform 0.12

Signed-off-by: baalajimaestro <me@baalajimaestro.me>

baalajimaestro committed:
Move provider to a file of itself

Signed-off-by: baalajimaestro <me@baalajimaestro.me>

baalajimaestro committed:
Bump up terraform and AWS requirements

Terraform - 1.2 and AWS 4.0

Signed-off-by: baalajimaestro <me@baalajimaestro.me>

baalajimaestro committed:
Update readme for bump in provider

Signed-off-by: baalajimaestro <me@baalajimaestro.me>

ashishmohite commented on main.tf:
```diff
@@ -174,7 +174,7 @@ resource "random_password" "password" {
 locals {
   cluster_identifier        = length(var.cluster_identifier) == 0 ? format("%s-%s-rds-cluster", var.service_name, var.environment) : var.cluster_identifier
-  tags                      = merge(var.tags, map("Name", format("%s-%s-rds-cluster", var.service_name, var.environment)))
+  tags                      = merge(var.tags, {"Name" = format("%s-%s-rds-cluster", var.service_name, var.environment)})
```
We shouldn't name any kind of RDS instance as a cluster

baalajimaestro committed:
Allow sensitive data to be exported for master_password

Signed-off-by: baalajimaestro <me@baalajimaestro.me>

baalajimaestro committed:
Dont use admin as the master account

admin is a reserved account and cannot be used as master on rds

Signed-off-by: baalajimaestro <me@baalajimaestro.me>

baalajimaestro committed:
Rework the variables

Most defaults are removed, considering how they are mandatory to be filled for the db to be created

Signed-off-by: baalajimaestro <me@baalajimaestro.me>
